test/functional/terminal/job_spec.lua

```diff
@@ -0,0 +1,71 @@
+local helpers = require('test.functional.helpers')(after_each)
+local thelpers = require('test.functional.terminal.helpers')
+local wait = helpers.wait
+
+describe('job control', function()
+  local screen
+  local script_file = 'Xtest_job_4569.vim'
+  local input_file = 'test/functional/fixtures/job_spec_124KB.txt'
+  local expected_bytes = '125769'
+
+  before_each(function()
+    helpers.clear()
+    screen = thelpers.screen_setup(0, '["'..helpers.nvim_prog..
+      '", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile"]')
+    screen:set_default_attr_ids({})
+    screen:set_default_attr_ignore(true)
+  end)
+  after_each(function()
+    os.remove(script_file)
+  end)
+
+  -- Note: This convoluted approach (spawned child nvim hosted in :terminal) is
+  -- the only way #4569 could be reproduced in a test. See also #4646.
+  it('job handler does not lose data #4569', function()
+    screen.timeout = 6000
+    helpers.write_file(script_file, [=[
+      let g:job_stdout = ''
+      function! s:JobHandler(job_id, data, event)
+        if 'stdout' ==# a:event
+          let g:job_stdout .= join(a:data,"\n")
+        elseif 'stderr' ==# a:event
+          let g:job_stdout = join(a:data,"\n")
+        endif
+      endfunction
+
+      let g:callbacks = {
+          \ 'on_stdout': function('s:JobHandler'),
+          \ 'on_stderr': function('s:JobHandler'),
+          \ 'on_exit'  : function('s:JobHandler')
+          \ }
+    ]=])
+    thelpers.feed_data(':source Xtest_job_4569.vim\n')
+
+    if helpers.os_name == 'windows' then
+      thelpers.feed_data(":call jobstart(['powershell', '-c', 'cat \""..input_file.."\"], g:callbacks)\n")
```

Comment:
powershell tends to have reasonable aliases to common unix shell commands (cat, grep, echo), so that should help with porting Windows tests.